### PR TITLE
Issue283

### LINKED ISSUE #283: Add "published" switch to posts & related stuff
- [x] "published" switch
- [x] only editable with given permission
- [x] not published  post are hidden by all endpoints
- [x] visible trough new endpoint
- [x] access token is requred for preview
- [x] preview link is provided for all posts
- [x] published posts can not be edited without permissions

### PR DESCRIPTION
**Pull request by Sasszem**
See issue desc.
Fixes #283

### CODE REVIEW

**Comment by Sasszem** [comment]
Build failing due to #285

**Comment by Sasszem** [comment]
Uuuups...
Touchscreen.

File: app/Models/Posts.php [hunk 1]
```diff
@@ -51,6 +52,28 @@ public function setImagesAttribute($value)
...
+    }
+
+    public function getPreviewLinkAttribute()
+    {
+        $id = $this->id;
+        $token = $this->previewToken;
+        if ($this->published) {
+            return "https://beta.verseghy-gimnazium.net/posts/$id";
+        } else {
+            return "https://beta.verseghy-gimnazium.net/posts/preview/$id&token=$token";
```
Comment (TwoDCube) [inline app/Models/Posts.php]: This should be `?token=`

Comment (Sasszem) [reply app/Models/Posts.php]: Done in 0308c54

**Pull request merged**